File: src/main/java/rag/trace/TraceEvent.java
package rag.trace;

public class TraceEvent {

    public final String stage;
    public final String summary;
    public final long durationMs;
    public final String error;

    public TraceEvent(String stage, String summary, long durationMs, String error) {
        this.stage = stage;
        this.summary = summary;
        this.durationMs = durationMs;
        this.error = error;
    }
}

File: src/main/python/rag/trace/trace_event.py
from typing import Optional
from dataclasses import dataclass
from rag.utils.timer_utils import now


@dataclass
class TraceEvent:
    stage: str
    summary: str
    duration_ms: int
    error: Optional[str]
    timestamp: int

    def __init__(self, stage: str, summary: str, duration_ms: int, error: Optional[str]) -> None:
        self.stage = stage
        self.summary = summary
        self.duration_ms = duration_ms
        self.error = error
        self.timestamp = now()






